File: 239_Sliding Window Maximum.py
from collections import deque

class Solution(object):
    def maxSlidingWindow(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        smax=list()
        if len(nums) == 0 or k ==0:
            return smax
        temp =deque(nums[:k])
        tmax = max(temp)
        smax.append(tmax)
        for i in range(k,len(nums)):
            temp.append(nums[i])
            if temp.popleft() == tmax or tmax < nums[i]:
                tmax = max(temp)
            smax.append(tmax)
        return smax
        

File: 239_Sliding Window Maximum.cpp

class Solution {
public:
    vector<int> maxSlidingWindow(vector<int>& nums, int k) {
        vector <int> maxel;
        int temp;
        if (nums.size() == 0)
            return nums;
        if (k == 1)
            return nums;
        temp = (*std::max_element(nums.begin(),nums.begin()+k));
        maxel.push_back(temp);
        for (int i=1;i<nums.size()-k+1;i++)
        {
            if (nums[i+k-1] >= temp)
            {
                temp = nums[i+k-1];
                maxel.push_back(temp);
                continue;
            }
            else if (nums[i-1] != temp)
            {
                maxel.push_back(temp);
                continue;
            }

            temp = (*std::max_element(nums.begin()+i,nums.begin()+i+k));   
            maxel.push_back(temp);
        }
     
        return maxel;   
    }
};